src/App.vue

```diff
@@ -71,11 +71,26 @@ function sendNoteOn(frequency: number, rawAttack: number) {
   return off
 }
 
-function midiNoteOn(index: number, rawAttack?: number) {
+function midiNoteOn(index: number, rawAttack?: number, channel?: number) {
+
+  // in multichannel-to-equave mode calculate an offset based on the incoming channel
+  if (midi.multichannelToEquave && channel !== undefined) {
+    let offset =  mmod(channel - midi.multichannelCenter + midi.multichannelEquavesDown, midi.multichannelNumEquaves) - midi.multichannelEquavesDown
+    offset = offset * scale.scale.size
+    index = index + offset
+  }
+  
   if (rawAttack === undefined) {
     rawAttack = 80
   }
+
   let frequency = scale.frequencies[index]
```

Comment:
Now that the `index` can get out of range `scale.getFrequency(index)` should be used unconditionally. (It uses pre-calculated freqs when available.)

Reply:
OK, will do.